### DISCUSSION TITLE
Docker Build Secret

### DISCUSSION THREAD

**Discussion by ThomasAunvik** [coollabsio/coolify#4294] [Feature Requests]
Allow using secrets trough environment variables and or file.

https://docs.docker.com/build/building/secrets/#using-build-secrets

Examples:

```
docker build --secret id=aws,src=$HOME/.aws/credentials .
```

```
docker build --secret id=kube,env=KUBECONFIG .
```

**Comment by tm1000** [comment]
I've discovered you can do this if you use docker compose as the builder and then use build secrets from environment

see: https://docs.docker.com/compose/how-tos/use-secrets/#build-secrets

**Comment by flawnn** [comment]
Further links regarding this:

- https://cryszon.github.io/posts/docker-build-secrets-in-coolify/
- https://docs.docker.com/engine/swarm/secrets/#use-secrets-in-compose
- https://stackoverflow.com/questions/42139605/how-do-you-manage-secret-values-with-docker-compose-v3-1
